File: dotnet/In3/Btc/TransactionInput.cs
using System.Text.Json.Serialization;

namespace In3.Btc
{
    /// <summary>
    /// Input of a transaction.
    /// </summary>
    public class TransactionInput
    {
        /// <summary>
        /// The transaction id.
        /// </summary>
        [JsonPropertyName("txid")] public string Txid { get; set; }

        /// <summary>
        /// The index of the transactionoutput.
        /// </summary>
        [JsonPropertyName("vout")] public uint Yout { get; set; }

        /// <summary>
        /// The script.
        /// </summary>
        [JsonPropertyName("scriptSig")] public ScriptSig ScriptSig { get; set; }

        /// <summary>
        /// Hex-encoded witness data (if any).
        /// </summary>
        [JsonPropertyName("txinwitness")] public string[] Txinwitness { get; set; }

        /// <summary>
        /// The script sequence number.
        /// </summary>
        [JsonPropertyName("sequence")] public uint Sequence { get; set; }
    }
}

File: java/src/in3/btc/TransactionInput.java
package in3.btc;

import in3.utils.JSON;

/**
 * Input of a transaction.
 */
public class TransactionInput {
  private JSON data;

  private TransactionInput(JSON data) {
    this.data = data;
  }

  protected static TransactionInput asTransactionInput(Object o) {
    if (o == null)
      return null;
    return new TransactionInput((JSON) o);
  }

  protected static TransactionInput[] asTransactionInputs(Object o) {
    if (o == null)
      return null;
    Object[] a           = (Object[]) o;
    TransactionInput[] b = new TransactionInput[a.length];
    for (int i = 0; i < a.length; i++)
      b[i] = TransactionInput.asTransactionInput(a[i]);
    return b;
  }

  /**
   * The transaction id.
   */
  public String getTxid() {
    return data.getString("txid");
  }

  /**
   * The index of the transactionoutput.
   */
  public long getYout() {
    return data.getLong("vout");
  }

  /**
   * The script.
   */
  public ScriptSig getScriptSig() {
    Object tx = (Object) data.get("scriptSig");
    if (tx == null)
      return null;

    return ScriptSig.asScriptSig(tx);
  }

  /**
   * Hex-encoded witness data (if any).
   */
  public String[] getTxinwitness() {
    return data.getStringArray("txinwitness");
  }

  /**
   * The script sequence number.
   */
  public long getSequence() {
    return data.getLong("sequence");
  }
}
